### DISCUSSION TITLE
problem with data_retention_days at schemas level

### DISCUSSION THREAD

**Discussion by qqcvd** [Snowflake-Labs/terraform-provider-snowflake#888] [Q&A]
Hi all,
I used the snowflake_schema  resource to create schemas.
the data_retention_days parameter is optional but if I omit the parameter, The Provider generate the parameter with the value set to 1 day.
Is it possible to forbid this behaviour ?
I woul like the data_retention_days at the schema level to be inheritate from the Database settings.

Best Regards

**Comment by thenaturalist** [comment]
Hi @qqcvd, 1 day is the default setting from Snowflake's side which is reflected in the provider. 
See `DATA_RETENTION_TIME_IN_DAYS` in the [CREATE SCHEMA](https://docs.snowflake.com/en/sql-reference/sql/create-schema.html) docs.

You can sure pass the value via Terraform code as well, either by a central local variable value, an output from a TF controlled DB or the [database data source](https://registry.terraform.io/providers/chanzuckerberg/snowflake/latest/docs/data-sources/database).
